File: baekjoon/StringQuestion/024-1543-문서검색.java
import java.io.BufferedReader;
import java.io.InputStreamReader;


public class Main {
    public int search(String src, String target) throws Exception {
        int count = 0;
        int begin = 0;
        while (begin + target.length() <= src.length()) {
            boolean matched = true;
            for (int i = 0; i < target.length(); i++) {
                if (src.charAt(begin + i) != target.charAt(i)) {
                    matched = false;
                    break;
                }
            }
            if (matched) {
                begin += target.length();
                count++;
            }
            else begin += 1;
        }

        return count;
    }

    public void soluton() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String src = br.readLine().replace("\n", "");
        String target = br.readLine().replace("\n", "");
        System.out.print(search(src, target));
    }

    public static void main(String[] args) throws Exception {
        new Main().soluton();
    }
}

File: baekjoon/StringQuestion/024-1543-문서검색.py
import sys; input = sys.stdin.readline


def search(src: str, target: str) -> int:
    cnt = 0
    begin = 0
    while begin + len(target) <= len(src):
        matched = True
        for i in range(len(target)):
            if src[begin + i] != target[i]:
                matched = False
                break
        if matched:
            begin += len(target)
            cnt += 1
        else: begin += 1
    
    return cnt


def main():
    src = input().rstrip()
    target = input().rstrip()
    print(search(src, target))


if __name__ == '__main__':
    main()